include/libopencm3/stm32/fdcan.h

```diff
@@ -0,0 +1,793 @@
+#include <libopencm3/cm3/common.h>
+
+/** @{ */
+
+/* FDCAN block base addresses. Used in functions to identify FDCAN block being manipulated. */
+
+/** @defgroup fdcan_block FDCAN block base addresses
+ * @{
+ */
+#define CAN1							FDCAN1_BASE
+#define CAN2							FDCAN2_BASE
+#define CAN3							FDCAN3_BASE
+/**@}*/
+
+
+/** @defgroup fdcan_fifo Named constants for FIFOs
+ * @{
+ */
+#define FDCAN_FIFO0						0
+#define FDCAN_FIFO1						1
+/**@}*/
+
+
+/** @defgroup FDCAN registers file in each FDCAN block. */
+
+#define FDCAN_CREL(can_base)			MMIO32(can_base + 0x0000)
+#define FDCAN_ENDN(can_base)			MMIO32(can_base + 0x0004)
+#define FDCAN_DBTP(can_base)			MMIO32(can_base + 0x000C)
+#define FDCAN_TEST(can_base)			MMIO32(can_base + 0x0010)
+#define FDCAN_RWD(can_base)				MMIO32(can_base + 0x0014)
+#define FDCAN_CCCR(can_base)			MMIO32(can_base + 0x0018)
+#define FDCAN_NBTP(can_base)			MMIO32(can_base + 0x001C)
+#define FDCAN_TSCC(can_base)			MMIO32(can_base + 0x0020)
+#define FDCAN_TSCV(can_base)			MMIO32(can_base + 0x0024)
+#define FDCAN_TOCC(can_base)			MMIO32(can_base + 0x0028)
+#define FDCAN_TOCV(can_base)			MMIO32(can_base + 0x002C)
+#define FDCAN_ECR(can_base)				MMIO32(can_base + 0x0040)
+#define FDCAN_PSR(can_base)				MMIO32(can_base + 0x0044)
+#define FDCAN_TDCR(can_base)			MMIO32(can_base + 0x0048)
+#define FDCAN_IR(can_base)				MMIO32(can_base + 0x0050)
+#define FDCAN_IE(can_base)				MMIO32(can_base + 0x0054)
+#define FDCAN_ILS(can_base)				MMIO32(can_base + 0x0058)
+#define FDCAN_ILE(can_base)				MMIO32(can_base + 0x005C)
+#define FDCAN_RXGFC(can_base)			MMIO32(can_base + 0x0080)
+#define FDCAN_XIDAM(can_base)			MMIO32(can_base + 0x0084)
+#define FDCAN_HPMS(can_base)			MMIO32(can_base + 0x0088)
+#define FDCAN_RXF0S(can_base)			MMIO32(can_base + 0x0090)
+#define FDCAN_RXF0A(can_base)			MMIO32(can_base + 0x0094)
+#define FDCAN_RXF1S(can_base)			MMIO32(can_base + 0x0098)
+#define FDCAN_RXF1A(can_base)			MMIO32(can_base + 0x009C)
+#define FDCAN_TXBC(can_base)			MMIO32(can_base + 0x00C0)
+#define FDCAN_TXFQS(can_base)			MMIO32(can_base + 0x00C4)
+#define FDCAN_TXBRP(can_base)			MMIO32(can_base + 0x00C8)
+#define FDCAN_TXBAR(can_base)			MMIO32(can_base + 0x00CC)
+#define FDCAN_TXBCR(can_base)			MMIO32(can_base + 0x00D0)
+#define FDCAN_TXBTO(can_base)			MMIO32(can_base + 0x00D4)
+#define FDCAN_TXBCF(can_base)			MMIO32(can_base + 0x00D8)
+#define FDCAN_TXBTIE(can_base)			MMIO32(can_base + 0x00DC)
+#define FDCAN_TXBCIE(can_base)			MMIO32(can_base + 0x00E0)
+#define FDCAN_TXEFS(can_base)			MMIO32(can_base + 0x00E4)
+#define FDCAN_TXEFA(can_base)			MMIO32(can_base + 0x00E8)
+#define FDCAN_CKDIV(can_base)			MMIO32(can_base + 0x0100)
+
+/* DAY[7:0]: FDCAN core revision date */
+#define FDCAN_CREL_DAY_SHIFT			0
+#define FDCAN_CREL_DAY_MASK				(0xFF << FDCAN_CREL_DAY_SHIFT)
+
+/* MON[7:0]: FDCAN core revision month */
+#define FDCAN_CREL_MON_SHIFT			8
+#define FDCAN_CREL_MON_MASK				(0xFF << FDCAN_CREL_MON_SHIFT)
```

Comment:
has this changed frrom last rev?  masks are normally specified unshifted, so they can be used by users to check their own values first if desired.  there are apis using pre-shifted, but all new stuff is unshifted, and other stuff slowly moves.  I'm sorry if I mised this the first time?

Reply:
I have removed use of __BITS() macro. I have probably took some older code with shifted masks as an example. Will adjust it as it definitely makes more sense.